### ISSUE TITLE
Cannot assign realm roles to user if user has all client roles

### ISSUE THREAD

**Issue by kelly-slice** [keycloak/keycloak#35994]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

If a user has all possible client roles additional realm roles cannot be assigned in the admin gui.  User does not have all realm roles assigned.  

### Version

26.0.7

### Regression

- [ ] The issue is a regression

### Expected behavior

Assign roles dialog box should show available realm roles.

### Actual behavior

Assign roles dialog box states "No roles for this user"  "There are no realm roles to assign"  The Filter and Search boxes do not appear.  

### How to Reproduce?

Assign all client roles to a user but not all realm roles.  Attempt to assign realm roles using the "Assign role" button.  Dialog box does not show available realm roles.  

### Anything else?

If even one client role is not currently assigned to the user the assign roles dialog box works as expected.

**Comment by ssilvert** [comment]
@kelly-slice I'm not sure I am following exactly how your user is configured.  Can you show me a list of all roles assigned when you see this problem.

**Comment by kelly-slice** [comment]
@ssilvert All client roles available in the realm are assigned to a user.

Steps to reproduce.  This was done with the stock 26.0.7 docker image. 

- Create a user
- Click "Assign role" under the "role mapping" tab.  Assign ALL client roles to the user.  In a stock 26.0.7 install they are as the list below.  If additional client roles have been added to the realm they must also be assigned.  
- Click "Assign role" again and are presented with "There are no realm roles to assign" even though the user only has all client roles.  The dialog should allow assignment of avalible realm roles.  

Roles assigned:
 	account delete-account
	account manage-account
	account manage-account-links
	account manage-consent
	account view-applications
	account view-consent
	account view-groups
	account view-profile
	broker read-token
	master-realm create-client
	master-realm impersonation
	master-realm manage-authorization
	master-realm manage-clients
	master-realm manage-events
	master-realm manage-identity-providers
	master-realm manage-realm
	master-realm manage-users
	master-realm query-clients
	master-realm query-groups
	master-realm query-realms
	master-realm query-users
	master-realm view-authorization
	master-realm view-clients
	master-realm view-events
	master-realm view-identity-providers
	master-realm view-realm
	master-realm view-users

**Comment by sschu** [comment]
@ssilvert From a usability-perspective, I don't even like the separation between client and realm roles in the UI. Why not list all roles and then let users search/filter?

**Comment by ssilvert** [comment]
@sschu I agree that we could do something better here.  The reason for the separation flows from the design of the server API.  But that's not really a good excuse.

Anyway, @kelly-slice, I understand the problem now.  Thanks for reporting it.

**Comment by edewit** [comment]
duplicate of #35311
